### PR TITLE
New folder Internal for internal header

### PR DESCRIPTION
**Pull request by LucileRemigy**
new folder "internal" in hpp/fcl for internal header + traits traversal for collision and distance

### CODE REVIEW

File: src/traits_traversal.h [hunk 1]
```diff
@@ -1,96 +1,91 @@
...
- *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
- *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
- *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
- *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- *  POSSIBILITY OF SUCH DAMAGE.
- */
-
-/** \author Florent Lamiraux */
-
-
```
Comment (jmirabel) [inline src/traits_traversal.h]: Would you mind adding the following header to this file:
```
/*
 * Software License Agreement (BSD License)
 *
 *  Copyright (c) 2019, CNRS-LAAS
 *  All rights reserved.
 */

/** \author Lucile Remigy, Joseph Mirabel */
```

File: src/traits_traversal.h [hunk 2]
```diff
@@ -1,96 +1,91 @@
...
+#include <../src/collision_node.h>
+#include <hpp/fcl/internal/traversal_node_setup.h>
+#include <../src/distance_func_matrix.h>
 
 namespace hpp
 {
 namespace fcl
 {
+
+#ifdef HPP_FCL_HAVE_OCTOMAP
```
Comment (jmirabel) [inline src/traits_traversal.h]: This pre-processor check if not at the correct place. It should only encapsulate OcTree template specialization.

Comment (LucileRemigy) [reply src/traits_traversal.h]: Do I delete the "ifdef hpp_fcl_have_octomap" ?

Comment (jmirabel) [reply src/traits_traversal.h]: No. The ifdef 
> should only encapsulate OcTree template specializations.

This means all the code like
```cpp
template <>
TraversalTraitsCollision<...OcTree...>
{
...
};
```

Comment (LucileRemigy) [reply src/traits_traversal.h]: is it better with the new push?

Comment (jmirabel) [reply src/traits_traversal.h]: Yes, perfect. Thank you.

**Pull request merged**